internal/cache/file/downloader/job.go

```diff
@@ -123,6 +127,26 @@ func (job *Job) init() {
 	job.status = JobStatus{NOT_STARTED, nil, 0}
 	job.subscribers = list.List{}
 	job.cancelCtx, job.cancelFunc = context.WithCancel(context.Background())
```

Comment:
Let's call the cancelFunc() when the operation completed successfully.

Ref: https://pkg.go.dev/context#WithCancel